doc/nrf/ug_radio_coex.rst

```diff
@@ -0,0 +1,196 @@
+
+   The ``*childImageName*`` string must be one of the following values:
+
+   *  ``multiprotocol_rpmsg`` for multiprotocol applications having support for both 802.15.4 and Bluetooth.
+   *  ``802154_rpmsg`` for applications having support for 802.15.4, but not for Bluetooth.
+   *  ``hci_rpmsg`` for application having support for Bluetooth, but not for 802.15.4.
+
+
+
+WiFi Coexistence implementations
+********************************
+
+The following CX implementations are available:
+
+* :ref:`Generic 3-pin interface`
+* :ref:`"Thread" implementaion`
+
+Each implementaion is a plugin consisting of one or more C source files interfacing with the MPSL API (which in turn communicates with the
+SR protocol drivers).
+
+When either CX variant is enabled, the driver requests access to the RF medium from the PTA and informs it about the operation it's about
+to perform. It also reacts properly to the information from the PTA that access to RF medium was granted/denied.
+
+Generic 3-pin interface
+=======================
+
+This implementation uses the following pins to communicate with the PTA:
+
+* REQUEST pin - output controlled by the protocol driver, it is asserted in order to request RF access from the PTA. It can be configured by
+  setting the `req-gpios` property of the `nrf_radio_coex` device tree node.
+* PRIORITY pin - output controlled by the protocol driver, it contains the information about the the type of operation
+  (RX/TX) to perform. It can be configured by setting the `pri-dir-gpios` property of the `nrf_radio_coex` device tree node.
+* GRANT pin - input of the SoC controlled by the PTA, asserts, when PTA grants access to RF to the 802.15.4 and deasserts when it denies the access. It can
+  be configured by setting the `grant-gpios` property of the `nrf_radio_coex` device tree node.
+
+The support for this interface is provided in a single-file plugin located in the sdk-nrf repo, in the subsys/mpsl/cx directory.
+It is only available to use with the Zephyr RTOS.
+
+Adding support for CX generic 3-pin interface
+---------------------------------------------
+
+An out of the box supported type of PTA is the so called "generic 3-pin interface".
+
+Generic 3-pin interface is supported out of the box. To use it complete the following steps:
+
+1. Set the :kconfig:`CONFIG_MPSL_CX_GENERIC_3PIN` Kconfig option to ``y``.
+
+2. Add the device tree `nrf_radio_coex` node in the way as described in :ref:`Hardware description`.
+
+After completing these steps no other function calls in the application are needed and the Coexistence will work.
+
+
+"Thread" implementaion
+======================
+
+This implementation is similar to the :ref:`Generic 3-pin interface`, however it is the REQUEST pin which contains information about the
+transfer direction and it is not multiplexed by using the PRIORITY pin (the device tree parameter `pri-dir-gpios` name is misleading here) This
```

Comment:
The generic 3-pin interface doesn't multiplex on the PRIORITY pin either.

Reply:
Not relevant anymore